lib/casmutils/xtal/structure.cxx

```diff
@@ -1,46 +1,158 @@
+#include <casm/crystallography/BasicStructure.hh>
+#include <casm/crystallography/SimpleStructure.hh>
+#include <casm/crystallography/SimpleStructureTools.hh>
+#include <casm/global/definitions.hh>
 #include <casmutils/definitions.hpp>
 #include <casmutils/exceptions.hpp>
 #include <casmutils/misc.hpp>
+#include <casmutils/xtal/coordinate.hpp>
 #include <casmutils/xtal/structure.hpp>
-
+#include <fstream>
 namespace extend
 {
 } // namespace extend
 
 namespace rewrap
 {
-Structure::Structure(const CASM::xtal::BasicStructure& init_struc) : structure_lattice(init_struc.lattice())
+Structure::Structure(const CASM::xtal::BasicStructure& init_struc)
+    : structure_lattice(init_struc.lattice()), casm_basicstructure(init_struc)
 {
-    throw except::NotImplemented();
+    _update_using("basic");
+}
+Structure::Structure(const CASM::xtal::SimpleStructure& init_struc)
+    : structure_lattice(init_struc.lat_column_mat), casm_simplestructure(init_struc)
+{
+    _update_using("simple");
 }
-
 Structure::Structure(const rewrap::Lattice& init_lat, const std::vector<rewrap::Site>& init_basis)
-    : structure_lattice(init_lat)
+    : structure_lattice(init_lat), basis(init_basis)
 {
-    throw except::NotImplemented();
-    /* for (const auto& site : init_basis) */
-    /* { */
-    /*     this->basis.push_back(site.__get()); */
-    /*     basis.back().set_lattice(this->lattice(), CASM::CART); */
-    /* } */
+    _update_using("internal");
 }
 
-Structure Structure::from_poscar(const fs::path& poscar_path) { throw except::NotImplemented(); }
+Structure Structure::from_poscar(const fs::path& poscar_path)
+{
+    CASM::xtal::BasicStructure pos;
+    if (!fs::exists(poscar_path))
+    {
+        throw except::UserInputMangle("Path does not exist");
```

Comment:
I realize we don't have documentation for the exceptions at the moment, but when I wrote this one it was meant for bad command line input. You can write a custom `BadPath` (or better name) that takes the path as an argument for this case, but generally it's fine to just throw a regular `std::runtime_error`, at least for the time being.